spec/dummy/app/views/components/modal.slim

```diff
@@ -0,0 +1,6 @@
+button#show Show
+
+= ui_component 'modal', \
```

Comment:
I don't think it's arbitrary, if you define the element and a reference to this element in the same place. We changed it anyway, in order to stick to the Bootstrap way.